modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -253,7 +254,26 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
         else if (backendId == DNN_BACKEND_WEBNN)
         {
             if (kernel_size.empty() || kernel_size.size() == 2)
-                return haveWebnn() && (type == MAX || type == AVE);
+            {
+                if (!haveWebnn())
+                {
+                    return false;
+                }
+                else
+                {
+                    if (!ceilMode)
+                    {
+                        CV_LOG_WARNING(NULL, "ceilMode is not supported by WebNN backend.");
+                        return false;
+                    }
+                    if (computeMaxIdx)
+                    {
+                        CV_LOG_WARNING(NULL, "Mask is not supported by WebNN backend.");
+                        return false;
+                    }
+                    return type == MAX || type == AVE; 
```

Comment:
Probably you can log the other type of pooling are not supported by WebNN backend.